### DISCUSSION TITLE
Removal of {{Interwiki} macro

### DISCUSSION THREAD

**Discussion by OnkarRuikar** [mdn/mdn-community#165] [Localization] [answered]
:wave:  All @mdn/localization-team-leads,

We have [removed](https://github.com/mdn/content/pull/18974) the `{{Interwiki}}` macro from `mdn/content`.
We would like to remove the macro from `mdn/translated-content` as well.

Following are the number of files, counted using command `rg -S interwiki -l | wc -l`:
```
es - 188
fr - 265
ja - 334
ko - 112
pt-br - 104
ru - 137
zh-cn - 194
zh-tw - 117
```
**Note:** not including `pl` and `de` locales, cos [they'll be archived soon](https://github.com/orgs/mdn/discussions/147).
***

There is no one-to-one mapping for `en` pages with other locales on Wikipedia. 
For example the macro `{{Interwiki("wikipedia", "Web accessibility")}}` on the [Accessibility (Deutsch)](https://developer.mozilla.org/de/docs/Glossary/Accessibility) page creates a broken link, <https://de.wikipedia.org/wiki/Web_accessibility>, at the moment. And in such cases the Wikipedia asks users to search the content in other languages. 
However, the English version <https://en.wikipedia.org/wiki/Web_accessibility> works fine.

Should such broken links be converted to English links?
If yes, then should the link text be localized or kept English?
For example,
```
[Web accessibility](https://en.wikipedia.org/wiki/Web_accessibility) 
or
[Web-Zugänglichkeit](https://en.wikipedia.org/wiki/Web_accessibility)
```
cc/ @teoli2003, @schalkneethling

**Comment by teoli2003** [comment]
cc/ @mdn/localization-team-leads

**Comment by SphinxKnight** [comment]
I created https://github.com/mdn/translated-content/issues/7287 for the different teams to track their work on the repo.

**Comment by marcelozarate** [comment]
I personally believe that since Wikipedia uses a URL naming that doesn't change, or automatically changes when changing locale at the URL, we could just replace the macro with the localized ones, when available, or the english ones when not. Tracking down the latter cases afterwards is an easy job as they all will be starting with `en.wiki(...)`

**Comment by OnkarRuikar** [comment]
> Should such broken links be converted to English links?

Yes. It's better than showing broken links.

***
> Should the link text be localized or kept English?

Show whatever text the macro is showing now. This way there will be no extra work required.
The macro's third parameter is custom link text, so let's use that or use the link part if not provided:
  `{{interwiki("wikipedia", "url text", "link text")}}` => `[link text](http://.../Url_text)`
  `{{interwiki("wikipedia", "url text")}}` => `[url text](http://.../Url_text)`

**Comment by Graywolf9** [comment]
`es` done in https://github.com/mdn/translated-content/pull/9818 and https://github.com/mdn/translated-content/pull/9819

**Comment by caugner** [comment, marked as answer]
The Interwiki macro has been removed in June 2023 via https://github.com/mdn/yari/pull/9264. 🎉
